Tests/iaas/image-metadata/image-md-check.py

```
@@ -282,12 +282,16 @@ def validate_imageMD(img, outd_list):
     if img.min_ram < 64:
         print(f'WARNING: Image "{imgnm}": min_ram == {img.min_ram} MiB < 64 MiB', file=sys.stderr)
         warnings += 1
-        # errors += 1
+    if not img.min_ram:
+        print(f'ERROR: Image "{imgnm}": min_ram == 0', file=sys.stderr)
+        errors += 1
     if img.min_disk * GIB < img.size:
```

One could argue that we should throw an error if
* if min_disk is 0 OR
* if min_disk is smaller than the image_size
```python
    if not img.min_disk or img.min_disk * GIB < img.size:
        print(f'ERROR: Image "{imgnm}": Too small min_disk == {img.min_disk * GiB/MiB:.0f} MiB vs. img_size == (img.size / MiB:.0f} MiB', file=sys.stderr)
        errors += 1
```
This would also be a hard rejection of too small sizes that can't be true and thus catch operator errors.

---

I agree that this would be good, but I suppose we would have to really change the standard for that. In the meantime, the warning could catch operator errors as well (if less effectively).

---

The other comment: We could add a sentence that the specified min_ram must be sufficient for the VM to boot and survive for 10s.
(Why 10s? Because scheduling latency plus boot time is in the order of a few seconds which makes it unreasonable to spawn VMs that only survive a single-digit number of seconds. You could do that with containers. If we ever wanted to create a test case, we also know that we don't need to wait very long. As a generic (OS-agnostic) health check is hard to create, we may never test for this. Maybe responding to arp requests is a valid OS-agnostic test? But of course we can hold providers accountable in case someone reports failure ...)

---

> I agree that this would be good, but I suppose we would have to really change the standard for that. In the meantime, the warning could catch operator errors as well (if less effectively).

It goes without saying that the metadata provided by the operator must reflect the truth and work, no?
So rejecting things hard that can't be true seems legit to me.